query_test.go

```diff
@@ -93,32 +91,26 @@ func TestQueryWatch(t *testing.T) {
 		t.Fatal("expected 0 peers")
 	}
 
-	// Start servers one at a time in the background.
-	go func() {
-		for i := 1; i <= nrPeers; i++ {
-			s, err := NewServer(etcd, ServerCfg{Namespace: namespace})
-			if err != nil {
-				t.Fatal(err)
-			}
+	for i := 1; i <= nrPeers; i++ {
```

Comment:
The outer goroutine and function here was somehow causing `defer etcd.Close()` to not fire. This then caused orphaned etcd entries to exist after the test, which meant if you re-ran the test within 30 seconds or so it'd fail because it wants to see zero peers to start. If it's ok to not start these in the background this is an easy fix.

If it's not, I'll probably need an adult.

Reply:
I meant `server.Close()`, not etcd